src/modules/Modal/Modal.js

```diff
@@ -144,16 +153,24 @@ class Modal extends Component {
     debug('handleDimmerClick()')
 
     e.stopPropagation()
-    this.onHide()
+
+    const { closeOnClickOutside } = this.props
+    if (closeOnClickOutside) {
+      this.onHide()
+    }
```

Comment:
This will still log a debug message and `e.stopPropagation()` even when `closeOnClickOutside` is false.  The condition should be added to short circuit early return at the top.  That way, nothing in the method runs if the condition is not met.